### PR TITLE
Alerting: various rule form fixes

### PR DESCRIPTION
**Pull request by domasx2**
* Rename "threshold alerts" and "system alerts" to "grafana managed alerts" and "cortex/loki managed alerts"
* Better validation for grafana timing fields: cannot be less than 10s, must be a multiple of 10s
* Fix issue with query editors submitting the form
* Various CSS fixes as per Oana's feedback
* Remove "keep last state" option from error state dropdowns
* Description, summary & runbook url annotation fields visible by default for new forms

https://user-images.githubusercontent.com/847684/118639168-469a3280-b7e0-11eb-9e96-f8fa0deaf22c.mp4

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by peterholmberg — COMMENTED** [review]
Just a small nit!

**Review by peterholmberg — APPROVED** [review]

**Comment by kylebrandt** [comment]
NoData should also be removed from the Alert state if execution error or timeout dropdown.

File: packages/grafana-ui/src/components/Forms/Field.tsx [hunk 1]
```diff
@@ -25,6 +25,9 @@ export interface FieldProps extends HTMLAttributes<HTMLDivElement> {
   error?: string | null;
   /** Indicates horizontal layout of the field */
   horizontal?: boolean;
+  /** maje validation message overflow horizontally. Prevents pushing out adjacent inline components */
```
[the remark this answers is not in the document]
Comment (domasx2) [reply packages/grafana-ui/src/components/Forms/Field.tsx]: :sweat_smile:

**Comment by domasx2** [comment]
> NoData should also be removed from the Alert state if execution error or timeout dropdown.

Thanks, updated!

**Pull request merged**